### PR TITLE
MainActivityViewModel & AuthorizationsListViewModel Tests

### PR DESCRIPTION
**Pull request by ConstantinKV**
- changed ConnectivityReceiver

### CODE REVIEW

File: app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+        assertThat(viewModel.onShowAuthorizationsListEvent.value, `is`(nullValue()))
+        assertThat(viewModel.onShowAuthorizationDetailsEvent.value, `is`(nullValue()))
+
+        //when
+        viewModel.onLifeCycleCreate(savedInstanceState, intent)
+
+        //then onShowAuthorizationsListEvent is posted
+        assertThat(viewModel.onShowAuthorizationsListEvent.value, equalTo(ViewModelEvent(Unit)))
+        assertThat(viewModel.onShowAuthorizationDetailsEvent.value, `is`(nullValue()))
+        assertThat(viewModel.onShowConnectEvent.value, `is`(nullValue()))
```
Comment (mnewlive) [inline app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt:101]: Mb smth like this : `assertNull(viewModel.onShowConnectEvent.value))` ?

File: app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+    @Test
+    @Throws(Exception::class)
+    fun onLifeCycleCreateTestCase1() {
+        /**
+         * given null savedInstanceState, null intent
+         */
+        val viewModel = createViewModel()
+        val savedInstanceState: Bundle? = null
+        val intent: Intent? = null
+        assertThat(viewModel.onShowAuthorizationsListEvent.value, `is`(nullValue()))
```
Comment (mnewlive) [inline app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt:92]: Add space before please

File: app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+    fun updateAppbarTestCase1() {
+        /**
+         * given appbar params
+         */
+        val viewModel = createViewModel()
+        val titleResId = R.string.app_name
+        val title = null
+        val backActionImageResId = null
+        val showMenu = true
+        assertThat(viewModel.appBarBackActionImageResource.value, equalTo(R.drawable.ic_appbar_action_back))
```
Comment (mnewlive) [inline app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt:582]: Add space please above

File: app/src/main/kotlin/com/saltedge/authenticator/app/ConnectivityReceiver.kt [hunk 4]
```diff
@@ -53,6 +56,11 @@ class ConnectivityReceiver : BroadcastReceiver() {
     }
 }
 
+interface ConnectivityReceiverAbs {
+    fun addNetworkStateChangeListener(listener: NetworkStateChangeListener)
+    fun removeNetworkStateChangeListener(listener: NetworkStateChangeListener)
```
[the remark this answers is not in the document]
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/app/ConnectivityReceiver.kt:61]: ConnectivityReceiver has collection of listener. And here i have added possibility to add and remove certain listener.

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt [hunk 5]
```diff
@@ -76,15 +76,40 @@ class AuthorizationsListViewModel @Inject constructor(
     val listItemUpdateEvent = MutableLiveData<ViewModelEvent<Int>>()
     val onConfirmErrorEvent = MutableLiveData<ViewModelEvent<String>>()
 
-    override fun onViewModelsExpired() {
-        listItemsValues.filter { it.shouldBeSetTimeOutMode }.forEach {
-            updateItemViewMode(listItem = it, newViewMode = ViewMode.TIME_OUT)
+    fun bindLifecycleObserver(lifecycle: Lifecycle) {
+        lifecycle.let {
+            it.removeObserver(this)
```
[the remark this answers is not in the document]
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt:81]: this call is designated to exclude adding of observer twice. i have noticed same call in some examples.

File: app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt (hunk 1 above)
Comment (ConstantinKV) [reply app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt:101]: i'd like to unify all assertions. assertNull belongs to anower library, but is not forbiden to use it. on your wish

File: app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt (hunk 2 above)
Comment (ConstantinKV) [reply app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt:92]: please clarify, before what i should add space?

Comment (mnewlive) [reply app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt:92]: yep, sorry
smth like this

```
        val viewModel = createViewModel()
        val savedInstanceState: Bundle? = null
        val intent: Intent? = null

        assertThat(viewModel.onShowAuthorizationsListEvent.value, `is`(nullValue()))
```

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (mnewlive) [reply app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt:81]: discussed

**Pull request merged**